src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,334 @@
+
+This script has 4 modes that can be executed independently if needed.
+
+ 1. Display big files specified by threshold value i.e. -t 5M. This option
+    will list files within provided local path with -f option and will prepend
+    each file with a \"Y\" if file exceeds threshold value and with \"N\" if
+    it does not.
+
+ 2. Compress files specified in provided local path with -f option that exceed
+    1M in size. This script will check if current local system is a VM host, if
+    so, compression will not be executed.
+
+ 3. Archive files specified in provided local path with -f option to provided
+    remote path destination.
+
+ 4. Create and display cart_logtest log files created by running cart_logtest.py
+    script. Output of running the script will be redirected to *_ctestlog.log
+    files.
+
+ Options:
+      -z        Compress files
+      -t        Threshold value to determine classification of big logs
+      -a        Remote path to scp files to i.e. server-A:/path/to/file
+      -f        Local path to files for archiving/compressing/size checking
+      -c        Enable running cart_logtest.py
+      -x        Enable performing compression when running on VM system
+      -v        Display commands being executed
+      -h        Display this help and exit
+"
+}
+
+#######################################
+# Check if the files exist on this host.
+# Arguments:
+#   $1: local logs to process
+# Returns:
+#   0: Files exist or no matches were found
+#   1: Missing the log files argument
+#######################################
+check_files_input() {
+    if [ -n "${1}" ]; then
+        # shellcheck disable=SC2086
+        if ls -d ${1} 2> /dev/null; then
+            echo "Files found that match ${1}."
+        else
+            echo "No files matched ${1}. Nothing to do."
+            exit 0
+        fi
+    else
+        echo "Please specify -f option."
+        exit 1
+    fi
+}
+
+#######################################
+# List and tag files with a 'Y' or 'N' to classify
+# big files based on the threshold provided.
+# Arguments:
+#   logs_dir: path to local logs
+#   threshold: value to use as threshold to classify big files
+# Returns:
+#   stdout output listing the files prepended with a 'Y' or 'N' tag
+#######################################
+list_tag_files() {
+    # shellcheck disable=SC2016,SC1004,SC2086
+    awk_cmd='{ \
+    if (th <= $1){ \
+        print "  Y:",$1,$2} \
+    else { \
+        print "  N:",$1,$2}\
+    }'
+
+    # Convert to bytes and run
+    bts=$(echo "${2}" | numfmt --from=iec)
+    # shellcheck disable=SC2086
+    du -ab -d 1 ${1} | awk -v th="${bts}" "${awk_cmd}"
+}
+
+#######################################
+# Check if this host is a VM.
+# Arguments:
+#   $1: whether or not to skip the check
+# Returns:
+#   0: VM detected
+#   1: No VM detected or skipped
+#######################################
+check_hw() {
+    if dmesg | grep -q "Hypervisor detected"; then
```

Comment:
I noticed the following error form `[2020-10-14T08:32:39.494Z] Running /usr/lib/daos/TESTING/ftest/get_remote_files.sh -z -x -v -f /var/tmp/ftest/avocado/job-results/latest/daos_logs/*.log*`:
```
[2020-10-14T08:32:39.495Z] Compressing files larger than 1M ...
[2020-10-14T08:32:39.495Z] + check_hw true
[2020-10-14T08:32:39.495Z] + dmesg
[2020-10-14T08:32:39.495Z] + grep -q 'Hypervisor detected'
[2020-10-14T08:32:39.495Z] dmesg: write failed: Broken pipe
[2020-10-14T08:32:39.495Z] dmesg: write error
[2020-10-14T08:32:39.495Z] + true
[2020-10-14T08:32:39.495Z] + echo '  Running compression on a VM host'
[2020-10-14T08:32:39.495Z]   Running compression on a VM host
[2020-10-14T08:32:39.495Z] + return 1
```
https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-2029/runs/54/nodes/616/log/?start=0

@brianjmurrell any thoughts on what might be causing this?

Reply:
This is expected, as get_remote_files.sh is being called with the -v option. With the verbose option on, it will turn on `set -ex` to output to the console which was the desired behavior, correct? I did see this issue and I am taking out the -v for this step as it is with the other steps.